docs/Producing effective SARIF.md

```diff
@@ -393,13 +393,25 @@ Similarly, most 'result' objects contain at least one 'artifactLocation' object.
 
 #### Messages
 
+##### `AvoidDuplicativeAnalysisTarget`: warning
+
+The 'analysisTarget' property '{1}' at '{0}' is unnecessary because it is the same as the result location. Remove the 'analysisTarget' property.
+
+#### `AvoidDuplicativeResultRuleInformation`: warning
+
+The result at '{0}' uses the 'rule' property to specify the violated rule, so it is not necessary also to specify 'ruleId' or 'ruleIndex'. Remove the 'ruleId' and 'ruleIndex' properties.
```

Comment:
You have a tendency to introduce descriptive prefixes in rules that I find distracting. The '{0}' value tends to be sufficiently descriptive to illuminate what's under discussion. This is most easily seen when actually reviewing output of the tool (which I encourage you to do), particularly real world quantities of output, not just a small # of test messages. In this second example. we see that we're trying to evolve to a common output format string for duplicative information (and we should converge on a standard).

'{0}' uses the 'rule' property to specify the violated rule, so it is not necessary also to specify 'ruleId' or 'ruleIndex'. This unnecessarily increases log file size. Remove the 'ruleId' and 'ruleIndex' properties.

Reply:
I'm open to this but I ask you to consider the potential value of the descriptive prefixes. The problem is that the JSON paths can be very long, and without the prefix, the first thing that meets your eye is this long string. Consider an example where it's _really_ long:

```
'/runs/0/results/0/locations/0/physicalLocation/artifactLocation/uri' contains an invalid URI 'fi:le:///C:/dev/'
```

I think this provides the reader with an easier entry to the message:

```
The URI 'fi:le:///C:/dev/' at '/runs/0/results/0/locations/0/physicalLocation/artifactLocation/uri' is invalid.
```

... because it says "I'm about to tell you something about a URI". Without the prefix, you have to scan to the end of the path string to learn that it's about a URI.

---
In reply to: [455907068](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455907068) [](ancestors = 455907068)

Reply:
Your example doesn't correspond to my feedback, which notes that describing the SARIF JSON type is generally not required because the type information tends to be encoded in the JSON path. The term 'URI' isn't typically encoded in the URI and so it is useful to identify the string literal as one.

---
In reply to: [455936290](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455936290) [](ancestors = 455936290,455907068)

Reply:
Ok, I understand the distinction.

---
In reply to: [455941928](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455941928) [](ancestors = 455941928,455936290,455907068)

Reply:
Just updated

---
In reply to: [455982124](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455982124) [](ancestors = 455982124,455941928,455936290,455907068)